core/coreunix/add.go

```diff
@@ -67,42 +67,46 @@ type AddedObject struct {
 	Bytes int64  `json:",omitempty"`
 }
 
-func NewAdder(ctx context.Context, n *core.IpfsNode, out chan interface{}) (*Adder, error) {
-	mr, err := mfs.NewRoot(ctx, n.DAG, newDirNode(), nil)
+func NewAdder(ctx context.Context, p pin.Pinner, bs bstore.GCBlockstore, ds dag.DAGService) (*Adder, error) {
```

Comment:
@jbenet this was changed because passing in a node gives us very little control over how the add gets done. Using a different blockstore or temporary dagservice for different applications is becoming fairly common.